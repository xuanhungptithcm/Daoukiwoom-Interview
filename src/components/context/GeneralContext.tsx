import { PropsWithChildren, ReactElement } from "react";
import ApiHandler from "../../api/ApiHandler";

import { GeneralDataContext, initialGeneralData } from "./constants";
import ContextProvider from "./ContextProvider";
import { IGenericProps } from "./types";

const GeneralContext = ({
  children,
}: PropsWithChildren<IGenericProps>): ReactElement => {
  return (
    <ContextProvider
      initialState={initialGeneralData}
      Context={GeneralDataContext}
    >
      <ApiHandler />
      {children}
    </ContextProvider>
  );
};

export default GeneralContext;
