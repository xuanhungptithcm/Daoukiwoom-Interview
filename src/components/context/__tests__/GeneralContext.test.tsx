/**
 * @jest-environment jsdom
 */

 import { shallow } from "enzyme";
 import GeneralContext from "../GeneralContext";

 jest.mock("../../../components/context/constants");

 describe("GeneralContext", () => {
   test("should render properly", () => {
     expect(shallow(<GeneralContext />)).toMatchSnapshot();
   });
 });
