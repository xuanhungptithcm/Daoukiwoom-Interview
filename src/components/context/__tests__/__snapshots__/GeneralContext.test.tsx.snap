// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralContext should render properly 1`] = `
<ContextProvider
  Context={
    Object {
      "$$typeof": Symbol(react.context),
      "Consumer": Object {
        "$$typeof": Symbol(react.context),
        "_context": [Circular],
      },
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": Array [],
      "_currentValue2": Array [],
      "_defaultValue": null,
      "_globalName": null,
      "_threadCount": 0,
    }
  }
  initialState={
    Object {
      "isLoggedIn": false,
      "userInfo": Object {
        "avatar": "",
        "createdAt": "",
        "id": "",
        "name": "",
        "password": "",
        "username": "",
      },
    }
  }
/>
`;
